### ISSUE TITLE
HLint ignores deriving inside family instances

### ISSUE THREAD

**Issue by mishun** [ndmitchell/hlint#215]
Working example:
```Haskell
{-# LANGUAGE GeneralizedNewtypeDeriving, TypeFamilies #-}
module Main where

import Control.Applicative

class Class x where
    data MyIO x :: * -> *

instance Class Int where
    newtype MyIO Int a = MyIO { unIO :: IO a }
        deriving (Functor, Applicative, Monad)

hello :: MyIO Int ()
hello = do
    MyIO $ putStrLn "Hello"
    MyIO $ putStrLn "World"

main :: IO ()
main = unIO hello
```
hlint-1.9.27 outputs:
```
Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE GeneralizedNewtypeDeriving, TypeFamilies #-}
Why not:
  {-# LANGUAGE TypeFamilies #-}

1 hint
```

**Comment by ndmitchell** [comment]
Thanks for the test case, all fixed now. I'll make a new release tomorrow.

**Comment by ndmitchell** [comment]
Released in hlint 1.9.28.
